cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/WidgetRuntimeService.kt

```diff
@@ -22,19 +24,18 @@ class WidgetRuntimeService(
     }
 
     fun deleteWidget(widgetConfig: JsonObject) {
-        val id = widgetConfig.getString(CogboardConstants.PROP_ID)
-        if (id != null) {
-            widgets.remove(id)?.stop()
-            LOGGER.info("Widget Deleted: $widgetConfig")
-        } else {
-            LOGGER.error("Widget Delete | " +
-                    "There is widget with no ID in configuration: $widgetConfig")
+        internalDelete("Delete", widgetConfig)
+    }
+
+    fun purgeWidget(widgetConfig: JsonObject) {
+        internalDelete("Purge", widgetConfig)?.let {
+            contentRepository.delete(it)
         }
     }
 
     fun createOrUpdateWidget(widgetConfig: JsonObject) {
         var newConfig = widgetConfig
-        val id = widgetConfig.getString(CogboardConstants.PROP_ID)
+        val id = extractId(widgetConfig)
```

Comment:
`widgetConfig.getId()` if you add extension function

Reply:
Good one, I'm still not fully used to Kotlin